notebooks/visualization/Declarative-Example.ipynb

```diff
@@ -0,0 +1,675 @@
+    "obs.time = datetime(1993, 3, 14, 0)\n",
+    "obs.level = 500 * units.hPa\n",
+    "obs.fields = ['temperature', 'dewpoint', 'height']\n",
+    "obs.locations = ['NW', 'SW', 'NE']\n",
+    "obs.formats = [None, None, lambda v: format(v, '.0f')[:3]]\n",
+    "obs.vector_field = ('u_wind', 'v_wind')\n",
+    "obs.reduce_points = 0\n",
+    "\n",
+    "# Add map features for the particular panel\n",
+    "panel = mpplots.MapPanel()\n",
+    "panel.layout = (1, 1, 1)\n",
+    "panel.area = (-124, -72, 20, 53)\n",
+    "panel.projection = 'lcc'\n",
+    "panel.layers = ['coastline', 'borders', 'states', 'land', 'ocean']\n",
+    "panel.plots = [obs]\n",
+    "\n",
+    "# Collecting panels for complete figure\n",
+    "pc = mpplots.PanelContainer()\n",
+    "pc.size = (15, 10)\n",
+    "pc.panels = [panel]\n",
+    "\n",
+    "# Showing the results\n",
+    "pc.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Model data\n",
+    "\n",
+    "Read in most recent NAM data from the Unidata Thredds Server Other model data can be found using the following website https://thredds.ucar.edu/thredds/catalog.html"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import data and remotely access it via Siphon using xarray\n",
+    "cat_string = 'https://thredds.ucar.edu/thredds/catalog/grib/NCEP/NAM/CONUS_12km/latest.xml'\n",
+    "cat = TDSCatalog(cat_string)\n",
+    "model_ds = cat.datasets[0].remote_access(use_xarray=True).metpy.parse_cf()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select the first time\n",
+    "model_ds = model_ds.sel(time = model_ds.time[0:1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Convert data to proper units using MetPy conversion tools\n",
+    "model_ds['Pressure_reduced_to_MSL_msl'] = model_ds.Pressure_reduced_to_MSL_msl.metpy.convert_units('hPa')\n",
+    "model_ds['Temperature_height_above_ground'] = model_ds.Temperature_height_above_ground.metpy.convert_units('degF')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Smooth the data using gaussian smoothing from SciPy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipy.ndimage import gaussian_filter\n",
+    "\n",
+    "# Smooth contour for MSLP\n",
+    "model_ds['Pressure_reduced_to_MSL_msl'] = (['time1', 'y','x'], gaussian_filter(model_ds['Pressure_reduced_to_MSL_msl'], sigma=3.0))\n",
+    "\n",
+    "# Smooth contour for surface temperature\n",
+    "model_ds['Temperature_height_above_ground'] = (['time1', 'height_above_ground1','y','x'], gaussian_filter(model_ds['Temperature_height_above_ground'], sigma=3.0))"
```

Comment:
If you move to metpy smooth_gaussian, you can get a DataArray out of that and plug it right back into the Dataset. I ran into an error because of the 'time1' variable changing.